[Merged by Bors] - feat(measure_theory/pi): `ae_eq` lemmas about intervals in `Π i, α i`

urkud opened the pull request:
---
- [x] depends on: #5685

urkud committed:
feat(measure_theory/pi): `ae_eq` lemmas about intervals in `Π i, α i`

fpvandoorn commented on src/data/set/lattice.lean:
```diff
@@ -1050,15 +1050,26 @@ instance : is_comm_applicative (set : Type u → Type u) :=
+variables {π : α → Type*}
+
+lemma pi_def (i : set α) (s : Πa, set (π a)) :
+  pi i s = (⋂ a∈i, eval a ⁻¹' s a) :=
```
```suggestion
  pi i s = (⋂ a ∈ i, eval a ⁻¹' s a) :=
```

fpvandoorn commented on src/measure_theory/pi.lean:
```diff
@@ -252,6 +252,75 @@ lemma pi_hyperplane [∀ i, sigma_finite (μ i)] (i : ι) [has_no_atoms (μ i)]
+variables {μ} [Π i, partial_order (α i)] [∀ i, sigma_finite (μ i)] [∀ i, has_no_atoms (μ i)]
+
+lemma pi_Iio_ae_eq_pi_Iic {s : set ι} {f : Π i, α i} :
+  pi s (λ i, Iio (f i)) =ᵐ[measure.pi μ] pi s (λ i, Iic (f i)) :=
```
Surely this can be factored through `(forall i, t i =ᵐ[\mu i] t' i) -> pi s t =ᵐ[measure.pi μ] pi s t'`? And the lemma with `forall` replaced by `exists` (or equivalently, dropped) is also true, right?

urkud replied:
The lemma with `exists` is not true: `[0, 1] × [0, 2]` is not the same as `[0, 1] × [0, 1]`.

fpvandoorn replied:
Good point, not sure what I was thinking there.

fpvandoorn commented on src/measure_theory/pi.lean:
```diff
@@ -252,6 +252,75 @@ lemma pi_hyperplane [∀ i, sigma_finite (μ i)] (i : ι) [has_no_atoms (μ i)]
+
+end intervals
+
+lemma pi_has_no_atoms (i : ι) [has_no_atoms (μ i)] [∀ i, sigma_finite (μ i)] :
```
Is there a reason for this lemma if we have the instance below it?

urkud replied:
```suggestion
/-- If one of the measures `μ i` has no atoms, them `measure.pi µ`
has no atoms. The instance below assumes that all `μ i` have no atoms. -/
lemma pi_has_no_atoms (i : ι) [has_no_atoms (μ i)] [∀ i, sigma_finite (μ i)] :
```

urkud committed:
Update src/measure_theory/pi.lean

urkud committed:
Snapshot

urkud committed:
Merge branch 'master' into pi-interval-ae-eq

urkud committed:
chore(order/filter/basic): a few `simp` lemmas

`order/filter/basic`

* add `filter.inter_mem_sets_iff`;
* rename `filter.Inter_mem_sets` to `filter.bInter_mem_sets`, make it
  an `iff` `[simp]` lemma;
* add a version `filter.bInter_finset_mem_sets` with a protected alias
  `finset.Inter_mem_sets`;
* rename `filter.sInter_mem_sets_of_finite` to
  `filter.sInter_mem_sets`, make it an `iff` `[simp]` lemma;
* rename `filter.Inter_mem_sets_of_fintype` to
  `filter.Inter_mem_sets`, make it an `iff `[simp]` lemma
* add `eventually` versions of the `*Inter_mem_sets` lemmas.

New `@[mono]` attributes

* `set.union_subset_union` and `set.inter_subset_inter` instead of
  `monotone_union` and `monotone_inter`; `mono*` failed to make a
  progress with `s ∩ t ⊆ s' ∩ t'` goal.
* `set.image2_subset`
* `closure_mono`

urkud committed:
Merge branch 'filter-simp-Inter' into pi-interval-ae-eq

urkud committed:
Use new `ae_eq_set_pi`

urkud committed:
chore(order/filter): a few more lemmas about `eventually` and set operations

urkud committed:
Merge branch 'master' into pi-interval-ae-eq
